File: builder.php
<?php

$lixeiras = [
    'quantidade' => 30,
    'capacidade' => 3,
    'tamanho' => 2,
];

$localizacoes = [ 'quantidade' => 15 ];

$habitacoes = [ 'quantidade' => 20 ];

$distanciaMaxima = 250;

// monta dados

$espacos = array_map(function ($localizacao, $i) {
    return $i . ' ' . mt_rand(4, 20);
}, $localizacoes, range(1, $localizacoes['quantidade']));

$volumes = array_map(function ($localizacao, $i) {
    return $i . ' ' . max(0, mt_rand(-4, 7));
}, $localizacoes, range(1, $localizacoes['quantidade']));

$capacidades = array_map(function ($i) use ($lixeiras) {
    return $i . ' ' . $lixeiras['capacidade'];
}, range(1, $lixeiras['quantidade']));

$tamanhos = array_map(function ($i) use ($lixeiras) {
    return $i . ' ' . $lixeiras['tamanho'];
}, range(1, $lixeiras['quantidade']));

$distancias = array_map(function ($i) use ($habitacoes) {
    $dados = array_map(function ($j) use ($i) {
        $multiplicador = 1 + abs($i - $j);
        return round(mt_rand(10, 30) * $multiplicador);
    }, range(1, $habitacoes['quantidade']));

    return "    " . $i . "\t\t" . implode("\t", $dados);
}, range(1, $localizacoes['quantidade']));

// imprime arquivo

$dados = "data;

param pn := " . $localizacoes['quantidade'] . ";
param ln := " . $lixeiras['quantidade'] . ";
param hn := " . $habitacoes['quantidade'] . ";

param e := " . implode(', ', $espacos) . ";
param v := " . implode(', ', $volumes) . ";
param c := " . implode(', ', $capacidades) . ";
param t := " . implode(', ', $tamanhos) . ";
param d :   " . implode("\t", range(1, $habitacoes['quantidade'])) . "   :=
" . implode("\n", $distancias) . ";
param DM := " . $distanciaMaxima . ";

end;
";

file_put_contents('teste6.mod', $dados);

File: builder.py
import random, json

bins = {
    'quantity': 205,
    'capacity': 1000,
    'size': 3,
}

locations = {
    'quantity': 458,
    'space': 30,
}

dwellings = { 'quantity': 801 }

distanceData = json.loads(open('./distances.json').read())
volumeData = json.loads(open('./volumes.json').read())

maxDistance = 300

# monta dados

spaces = map(
    lambda i: '%s %s' % (i, locations['space']),
    range(1, locations['quantity'] + 1)
)

capacities = map(
    lambda i: '%s %s' % (i, bins['capacity']),
    range(1, bins['quantity'] + 1)
)

sizes = map(
    lambda i: '%s %s' % (i, bins['size']),
    range(1, bins['quantity'] + 1)
)

volumes = map(
    lambda i: '%s %s' % (i, round(volumeData[i - 1])),
    range(1, dwellings['quantity'] + 1)
)

distances = map(
    lambda i: '    %s\t\t%s' % (i, '\t'.join(map(
        lambda j: str(round(distanceData[str(i - 1)][str(j - 1)])),
        range(1, dwellings['quantity'] + 1)
    ))),
    range(1, locations['quantity'] + 1)
)

dwellingsString = map(
    lambda i: str(i),
    range(1, dwellings['quantity'] + 1)
)

# imprime arquivo

file = open('teste.mod', 'w')
file.write('''data;

param pn := %s;
param ln := %s;
param hn := %s;

param e := %s;
param c := %s;
param t := %s;
param v := %s;
param d :   %s   :=
%s;
param DM %s;

end;
''' % (
    locations['quantity'],
    bins['quantity'],
    dwellings['quantity'],
    ', '.join(spaces),
    ', '.join(capacities),
    ', '.join(sizes),
    ', '.join(volumes),
    '\t'.join(dwellingsString),
    '\n'.join(distances),
    maxDistance
))
file.close()
